### PR TITLE
LPS-130366 Migrate selection modals in commerce-product-type-grouped-web

### PR DESCRIPTION
**Pull request by kresimir-coko**
This is a part of an [ongoing epic](https://issues.liferay.com/browse/LPS-129246) to replace legacy AUI-based modals with Clay/React-based `openSelectionModal`. In this PR, we are replacing modals in `commerce-product-type-grouped-web`.

JIRA: https://issues.liferay.com/browse/LPS-130366

Previously reviewed at https://github.com/liferay-frontend/liferay-portal/pull/1049

# 🧪 Testing
1. Product menu > Commerce > Products
2. Add new `Grouped` product
3. Go to `Grouped` tab and click on `+`

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp [hunk 1]
```diff
@@ -210,39 +214,40 @@ renderResponse.setTitle(cpDefinition.getName(themeDisplay.getLanguageId()));
...
-						var <portlet:namespace />addCPDefinitionIds = [];
+	if (addDefinitionGroupedEntryButton) {
+		addDefinitionGroupedEntryButton.addEventListener('click', (event) => {
+			event.preventDefault();
 
-						var selectedItems = event.newVal;
+			Liferay.Util.openSelectionModal({
+				multiple: true,
+				onSelect: function (selectedItems) {
+					if (selectedItems?.length) {
```
Comment (markocikos) [inline modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:227]: We cannot use optional chaining in JSPs, unfortunately. That is what source formatter is complaining about. Here is some strange unknown guy with more details: https://github.com/liferay-frontend/liferay-portal/pull/1015#issuecomment-832767394

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp [hunk 2]
```diff
@@ -210,39 +214,40 @@ renderResponse.setTitle(cpDefinition.getName(themeDisplay.getLanguageId()));
...
+							'<portlet:namespace />addCPDefinitionGroupedEntryFm'
+						);
 
+						if (addCPDefinitionGroupedEntryFm) {
 							submitForm(addCPDefinitionGroupedEntryFm);
 						}
-					},
+					}
 				},
+				selectEventName: 'productDefinitionsSelectItem',
```
Comment (markocikos) [inline modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:245]: If this is a standard multiple selection, we can remove `selectEventName`

**Review by markocikos — COMMENTED** [review]
@kresimir-coko Minor bits inline. Other than that, looks good!

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:227]: 👍

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp (hunk 2 above)
Comment (kresimir-coko) [reply modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:245]: 👍

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
> ❌ ci:test:sf - 0 out of 1 jobs passed in 4 minutes
```
[exec] > Task :packageRunCheckFormat
[exec] yarn run v1.13.0
[exec] \$ liferay-npm-scripts check
[exec] apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp: BAD
[exec] Prettier checked 1 file, 1 file has problems
```

I can't figure out why this is failing, I can't find how I can triangulate the issue, locally it works 🤷

**Comment by markocikos** [comment]
> can't figure out why this is failing, I can't find how I can triangulate the issue, locally it works

@kresimir-coko Before sending PRs, make sure you run portal's source formatter:
1. go to `portal-impl`
2. run `ant format-source-current-branch`

It does some auto-formatting on this PR.

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
@FabioDiegoMastrorilli can you please get this one reviewed? Let me know if I should rebase with master first.

**Comment by kresimir-coko** [comment]
@FabioDiegoMastrorilli bump

**Comment by markocikos** [comment]
@FabioDiegoMastrorilli a gentle reminder

**Comment by kresimir-coko** [comment]
I have rebased with master and force-pushed @FabioDiegoMastrorilli. Please review.

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp [hunk 3]
```diff
@@ -210,39 +214,43 @@ renderResponse.setTitle(cpDefinition.getName(themeDisplay.getLanguageId()));
...
+							entryCPDefinitionIds.value = selectedItems.map(
+								(item) => item.value
 							);
+						}
+
+						const addCPDefinitionGroupedEntryFm = document.getElementById(
+							'<portlet:namespace />addCPDefinitionGroupedEntryFm'
+						);
 
+						if (addCPDefinitionGroupedEntryFm) {
```
Comment (FabioDiegoMastrorilli) [inline modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:244]: @kresimir-coko this looks overly defensive. `addCPDefinitionGroupedEntryFm` is not conditionally rendered so, when undefined, throwing an error should be more correct (it also gives more hints for fixing).

(It's absolute not a major thing, just a personal thought so feel free not to update it)

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp [hunk 4]
```diff
@@ -190,7 +190,11 @@ renderResponse.setTitle(cpDefinition.getName(themeDisplay.getLanguageId()));
 				'<liferay-ui:message key="are-you-sure-you-want-to-delete-the-selected-entries" />'
 			)
 		) {
-			var form = window.document['<portlet:namespace />fm'];
```
Comment (FabioDiegoMastrorilli) [inline modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:193]: First time in my life I see this :O

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp (hunk 3 above)
Comment (kresimir-coko) [reply modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:244]: We shouldn't delay this any further I think, so I won't change it, for now 👍

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp (hunk 4 above)
Comment (markocikos) [reply modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:193]: It is an old custom API. It's not a blocker, but we should use `getElementById` instead. See more details in https://github.com/liferay-frontend/liferay-portal/pull/757#discussion_r570805703

File: modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp (hunk 3 above)
Comment (FabioDiegoMastrorilli) [reply modules/apps/commerce/commerce-product-type-grouped-web/src/main/resources/META-INF/resources/view_cp_definition_grouped_entries.jsp:244]: Totally ok ;)

**Comment by FabioDiegoMastrorilli** [comment]
ci:forward

**Pull request closed without merging**